docs/releases/slashing-release/interfaces/slashing-parameter-configuration.md

```diff
@@ -1,176 +1,103 @@
-     * @dev reverts if strategies.length != nonslashableToAdd.length
-     * @dev emits event TotalAndNonSlashableMagnitudeUpdated
+     * @param operator address to decrease allocations for
+     * @param deallocations array of magnitude adjustments for multiple strategies and corresponding operator sets
+     * @param operatorSignature signature of the operator if msg.sender is not the operator
      */
-    function queueMagnitudeDilution(
+    function queueDeallocation(
         address operator,
-        IStrategy[] calldata strategies,
-        uint64[] calldata nonslashableToAdd,
-        SignatureWithSaltAndExpiry calldata allocatorSignature
-    ) external returns (uint64 newNonslashableMagnitude, uint64 newTotalMagnitude);
+        MagnitudeAdjustment[] calldata deallocations,
+        SignatureWithSaltAndExpiry calldata operatorSignature
+    ) external;
 
     /**
-     * @notice Decrement the nonslashable magnitude to increase all operatorSet slashable stake allocations
-     * proportionally. This is an efficient way of adjusting total magnitude and slashing risk.
+     * @notice Complete queued deallocations of slashable stake for an operator.
+     * Increments the nonslashable magnitude of the operator by the sum of all deallocation amounts for each strategy. 
+     * If the operator was slashed, this will be a smaller amount than during queuing.
      *
-     * @param operator the address that will have their nonslashable magnitude decremented, increasing
-     * relative proportions of all existing operatorSet magnitudes
-     * @param strategies array of IStrategy's to dilute
-     * @param nonslashableToDecrement array values of the magnitude amount decremented from the nonslashable magnitude
-     * @param allocatorSignature if non-empty is the signature of the allocator on
-     * the modification. if empty, the msg.sender must be the allocator for the
-     * operator
+     * @param operator address to complete deallocations for
+     * @param operatorSets the sets of operator sets to complete deallocations for
+     * @param strategies the list of strategies to complete deallocations for, must be 1:1 with the operatorSets
+     * @param operatorSignature signature of the operator if msg.sender is not the operator
      *
-     * @dev reverts if nonslashableToDecrement > latest pending nonslashableMagnitude
-     * @dev reverts if there are more than 6 pending changes to changes to the nonslashable magnitude for the strategy
-     * @dev reverts if strategies.length != nonslashableToDecrement.length
-     * @dev emits event TotalAndNonSlashableMagnitudeUpdated
+     * @dev can be called permissionlessly by anyone
+     * @dev must have the same number of operatorSets and strategies
      */
-    function queueMagnitudeConcentration(
+    function completeDeallocations(
         address operator,
+        OperatorSet[] calldata operatorSets,
```

Comment:
Looks like there will be duplicate strategies passed in here. Should opSets be 2-D array or a special struct for this?

Reply:
good point